app/controllers/fhir/r4/api_controller.rb

```diff
@@ -786,6 +786,8 @@ def format_model_validation_errors(resource)
       value = resource[attribute] || resource.public_send("#{attribute}_before_type_cast")
       msg_header = 'Validation Error' + (value ? " for value '#{value}'" : '') + " on '#{VALIDATION[attribute][:label]}':"
       errors.each do |error_message|
+        # Exclude the actual value in logging to avoid PII/PHI
+        Rails.logger.info("Validation Error on #{attribute}: #{error_message}")
```

Comment:
I think this is the only place potentially sensitive information could fall into, depending on the error message. I need to investigate some more.

Reply:
Yeah I hear you. I gave it some thought and I couldn't see any way this could end up logging sensitive info. It could give you some vague information about the info, e.g. some values that it's _not_, but I don't think it would ever log the actual value. However, I don't love that that would be contingent on error messages keeping to that truth in the future. If you'd prefer, I would be fine reducing this to just logging the attribute name. It doesn't give us as much information of course, but it's still useful, especially because most individual attributes only have one validation rule. Just let me know your preference.

Reply:
This could end up logging actual values for attributes on the Patient. I'm not sure what we define as "sensitive", but one example would be if the Patient had an telephone number when their preferred contact method was one of 'SMS Texted Weblink', 'Telephone call', or 'SMS Text-message'. The validation message would say something like:
```
is required when 'Primary Contact Method' is 'SMS Text-message' 
```
revealing the value of `preferred_contact_method`.

Other examples include error messages for `symptom_onset` or `last_date_of_exposure` revealing which workflow the Patient is in. Again not sure if this all is "sensitive", but it could be logged.

Reply:
Ah yeah good call. That's enough for me. I'm going to remove the logging of the message

Reply:
https://github.com/SaraAlert/SaraAlert/pull/707/commits/cd4f0914dbafee04c9541c5aef5369b9340802ba